Backend/src/ml_lib/shared/shared_model.py

```diff
@@ -32,11 +32,11 @@ def run(self, file_name: str, is_first: bool) -> None:
         if not is_first:
             df = self.latter_run(data)
         else:
-            df = self.first_run(data)
+            df = self.first_run(data, dim)
 
         self._df_setter(df)
 
-    def first_run(self, df: pd.DataFrame) -> pd.DataFrame:
+    def first_run(self, df: pd.DataFrame, dim: str = None) -> pd.DataFrame:

```

Comment:
The type for dim should be `str | None` here as well, sorry I missed this on the last review. Same should apply to the sub_models

Reply:
done